index.js

```diff
@@ -786,7 +786,10 @@
 
   //  Array$prototype$extend :: Array a ~> (Array a -> b) -> Array b
   function Array$prototype$extend(f) {
-    return [f(this)];
+    var xs = this;
+    return xs.map(function(_, i) {
+      return f(xs.slice(i));
+    });
```

Comment:
@safareli may insist on avoiding all this slicing. :hocho: :watermelon:

If he does we could use this approach:

```javascript
var result = [];
for (var xs = this.slice(); xs.length > 0; xs.shift()) result.push(f(xs));
return result;
```